src/microbe_stage/SpawnSystem.cs

```diff
@@ -68,9 +68,12 @@ public class SpawnSystem
     /// </summary>
     private int estimateEntityCount;
 
+    private Dictionary<Spawner, int> attemptsPerSpawnType;
```

Comment:
wouldn't it be better to have the attempts counter in the `Spawner` class itself? Or is there somthing prohibiting this I don't see?

Reply:
No, you are right, the spawner-dependent part could be stored in `Spawner` indeed for memory saving.

However, given that I think I'm finally succeeding with a more efficient rework with PR #2397 (which also solves this spawn order thing), I think I'll discontinue this one for a bit, and only resume if the other PR actually does not work / takes longer and a hotfix is needed.